model/project_ref.go

```diff
@@ -1255,6 +1315,176 @@ func (projectRef *ProjectRef) Upsert() error {
+			})
+	case ProjectRefWorkstationsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{projectRefWorkstationConfigKey: p.WorkstationConfig},
+			})
+	case ProjectRefTriggersSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					projectRefTriggersKey: p.Triggers,
+				},
+			})
+	case ProjectRefAliasSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{
+					projectRefPatchTriggerAliasesKey: p.PatchTriggerAliases,
+				},
+			})
+	case ProjectRefPeriodicBuildsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$set": bson.M{projectRefPeriodicBuildsKey: p.PeriodicBuilds},
+			})
+	case ProjectRefVariablesSection:
+		// this section doesn't modify the project ref
+		return false, nil
+	default:
+		return false, errors.Errorf("invalid section")
+	}
+
+	if err != nil {
+		return false, errors.Wrapf(err, "error defaulting to repo for section '%s'", ProjectRefGithubAndCQSection)
+	}
+	return true, nil
+}
+
+// DefaultSectionToRepo modifies a subset of the project ref to use the repo values instead.
+// This subset is based on the pages used in Spruce.
+// If project settings aren't given, we should assume we're defaulting to repo and we need
+// to create our own project settings event  after completing the update.
+func DefaultSectionToRepo(projectId string, section ProjectRefSection, userId string) error {
+	before, err := GetProjectSettingsEventById(projectId)
+	if err != nil {
+		return errors.Wrap(err, "error getting before project settings event")
+	}
+
+	modified, err := saveProjectRefForSection(projectId, nil, section)
+	if err != nil {
+		return errors.Wrapf(err, "error defaulting project ref to repo for section '%s'", section)
+	}
+
+	// Handle sections that modify collections outside of the project ref.
+	// Handle errors at the end so that we can still log the project as modified, if applicable.
+	catcher := grip.NewBasicCatcher()
+	switch section {
+	case ProjectRefVariablesSection:
+		err = db.Update(ProjectVarsCollection,
+			bson.M{ProjectRefIdKey: projectId},
+			bson.M{
+				"$unset": bson.M{
+					projectVarsMapKey:    1,
+					privateVarsMapKey:    1,
+					restrictedVarsMapKey: 1,
+				},
+			})
+
+		catcher.Wrapf(err, "error defaulting to repo for section '%s'", section)
+	case ProjectRefGithubAndCQSection:
+		for _, a := range before.Aliases {
+			// remove only internal aliases
+			if utility.StringSliceContains(evergreen.InternalAliases, a.Alias) {
+				catcher.Add(RemoveProjectAlias(a.ID.Hex()))
+			}
+		}
+		if catcher.HasErrors() {
+			// log the changes before returning whatever errors were caught
+			_ = getAndLogProjectModified(projectId, userId, before)
+			return errors.Wrapf(catcher.Resolve(), "error defaulting to repo for section '%s'", section)
```

Comment:
Do we maybe want to remove the return from here and let the return at the end of the function handle it, since that resolves the catcher as well?

Reply:
yep I did this wrong, fixed